services/tables/src/main/java/com/linkedin/openhouse/tables/utils/TableUUIDGenerator.java

```diff
@@ -38,6 +36,8 @@ public class TableUUIDGenerator {
 
   @Autowired StorageManager storageManager;
 
+  @Autowired OpenHouseInternalCatalog catalog;
```

Comment:
Lets avoid adding `OpenHouseInternalCatalog` at this layer and resolve this purely using `Storage` apis. Reasoning: We may onboard non-iceberg table specs, and internalcatalog introduces only-iceberg semantics.

Reply:
+1

Reply:
done